app/lib/scorecard/backend.dart

```diff
@@ -0,0 +1,178 @@
+
+/// Sets the active scorecard backend.
+void registerScoreCardBackend(ScoreCardBackend backend) =>
+    ss.register(#_scorecard_backend, backend);
+
+/// The active job backend.
+ScoreCardBackend get scoreCardBackend =>
+    ss.lookup(#_scorecard_backend) as ScoreCardBackend;
+
+class ScoreCardBackend {
+  final db.DatastoreDB _db;
+  ScoreCardBackend(this._db);
+
+  Future<ScoreCardData> getScoreCardData(
+    String packageName,
+    String packageVersion, {
+    @required bool onlyCurrent,
+  }) async {
+    final cached = await scoreCardMemcache.getScoreCardData(
+        packageName, packageVersion, versions.runtimeVersion,
+        onlyCurrent: onlyCurrent);
+    if (cached != null) {
+      return cached;
+    }
+
+    final key = scoreCardKey(packageName, packageVersion);
+    final currentList = await _db.lookup([key]);
+    if (currentList.first != null) {
+      final data = (currentList.first as ScoreCard).toData();
+      await scoreCardMemcache.setScoreCardData(data);
+      return data;
+    }
+
+    if (onlyCurrent) {
+      return null;
+    }
+
+    final query = _db.query<ScoreCard>(ancestorKey: key.parent)
+      ..filter('<', versions.runtimeVersion);
+    final all = await query
+        .run()
+        .where((sc) =>
+            // sanity check to not rely entirely on the lexicographical order
+            isNewer(sc.semanticRuntimeVersion, versions.semanticRuntimeVersion))
+        .toList();
+    if (all.isEmpty) {
+      return null;
+    }
+    all.sort((a, b) =>
+        isNewer(a.semanticRuntimeVersion, b.semanticRuntimeVersion) ? -1 : 1);
+    final data = all.last.toData();
+    await scoreCardMemcache.setScoreCardData(data);
+    return data;
+  }
+
+  Future updateReport(
+      String packageName, String packageVersion, ReportData data) async {
+    final key = scoreCardKey(packageName, packageVersion)
+        .append(ScoreCardReport, id: data.reportType);
+    await _db.withTransaction((tx) async {
+      ScoreCardReport report;
+      final reportList = await tx.lookup([key]);
+      report = reportList.first as ScoreCardReport;
+      if (report != null) {
+        _logger.info(
+            'Updating report: $packageName $packageVersion ${data.reportType}.');
+        report
+          ..updated = new DateTime.now().toUtc()
+          ..reportStatus = data.reportStatus
+          ..reportJson = data.toJson();
+      } else {
+        _logger.info(
+            'Creating new report: $packageName $packageVersion ${data.reportType}.');
+        report = new ScoreCardReport.init(
+          packageName: packageName,
+          packageVersion: packageVersion,
+          reportData: data,
+        );
+      }
+      tx.queueMutations(inserts: [report]);
+      await tx.commit();
+    });
+  }
+
+  Future<Map<String, ReportData>> loadReports(
+      String packageName, String packageVersion,
+      {List<String> reportTypes}) async {
+    reportTypes ??= [ReportType.pana, ReportType.dartdoc];
```

Comment:
These are static const, can you set a default parameter value for reportTypes instead, if the list can be const?